src/autocomplete.js

```diff
@@ -165,7 +164,7 @@ export default class Autocomplete extends Component {
       menuOpen: newState.menuOpen || false,
       query: newQuery,
       selected: null
-    })
+    }, this.props.confirmOnBlur ? this.props.onConfirm(newQuery) : null)
```

Comment:
> I opted to just send `newQuery`

👍  That's fine. `options[selected]` was weird given that `newQuery` exists in the same scope.

The second argument to `setState` should be a callback function. You're passing the result of the `onConfirm()` invocation. So your code (when `this.props.confirmOnBlur === true`) at runtime is:

```js
this.setState({
  focused: null,
  menuOpen: newState.menuOpen || false,
  query: newQuery,
  selected: null
}, undefined)
```

With the added side effect of running `onConfirm` in order to produce the `undefined` argument for `setState`, which shouldn't actually solve the underlying issue (if it did, that's curious).

Instead you want this (added `() =>`):

```js
this.setState({
  focused: null,
  menuOpen: newState.menuOpen || false,
  query: newQuery,
  selected: null
}, () => this.props.confirmOnBlur ? this.props.onConfirm(newQuery) : null)